client/app/services/group/currentGroup.service.js

```diff
@@ -9,18 +9,39 @@
 */
 export default class CurrentGroup {
 
-  constructor() {
+  constructor(User, Authentication) {
+    "ngInject";
     Object.assign(this, {
-      value: {}
+      value: {},
+      User,
+      Authentication
     });
   }
 
   set(value) {
     angular.copy(value, this.value);
+    if (angular.isUndefined(this.Authentication.data)) {
+      this.Authentication.update();
```

Comment:
This is wrong. The update() method sets `Authentication.data` asynchronously. If the authentication data does not exist, `persistCurrentGroup` will use the undefined value before it gets updated.

I think we can assume that setting the current group is only possible when a user is logged in. Therefore, the call to Authentication.update() should not be necessary.
At the maximum, I would check in persistCurrentGroup that `Authentication.data` is defined and throw an error if not. But I'm also happy to leave this check away.